src/content/blog/2025_ucc_unitaryhack.md

```diff
@@ -0,0 +1,136 @@
+
+<div style="display: flex; flex-direction: column; align-items: center;">
+    <img style="margin-top: 0; margin-bottom: 0;" src="/images/2025_ucc_unitaryhack/MPS-cartoon-pennylane.png" alt="cartoon of a Matrix Product State (MPS) drawn as a strip of paper with scissors poised to cut at regular intervals" width="200"/>
+    <span style="font-size: 0.9em; color: #666; margin-top: 4px; display: block; text-align: center;">
+        Source: <a href="https://pennylane.ai/qml/demos/tutorial_mps" target="_blank" style="color: #666; text-decoration: underline;">PennyLane MPS tutorial</a>
+    </span>
+</div>
+
+**Challenges I faced in hacking on this issue:**
+While developing the open-source [qmprs](https://github.com/Qualition/qmprs) project for compression and compilation of quantum circuits using tensor networks, I had become familiar with the side-effects of approximation in state encoding. However, it was through **UCC’s suggestion to benchmark the compiler using their suite that a critical limitation came to light**—random Clifford circuits revealed themselves as a particular weakness of MPS-based approaches. Digging into the entanglement properties of both the benchmark circuits and the typical states I had used, I realized the circuits exhibited volume-law entanglement, whereas the states I had relied on were coincidentally area-law entangled. This contrast helped explain the discrepancy and highlighted both an opportunity and a constraint for the proposed compiler.  
+I spent time during and after the event exploring ways to reconcile this, eventually incorporating a method to ensure consistency for volume-law circuits via variational fine-tuning. 
+
+**The experience was deeply instructive**, and I was fortunate to contribute to the project, though it was UCC’s benchmarking approach that uncovered the limitation in the first place—a contribution I remain genuinely thankful for.
+Additionally, UCC’s infrastructure emphasized the importance of compatibility between approximate and exact compilation methods. To support this, I introduced a fallback mechanism that checks fidelity and cost metrics, defaulting to UCC’s exact compiler when approximation underperforms. It was a straightforward solution, but in hindsight, one that proved to be quite practical.
+
+**Future projects I'm interested in working on:**
+My current focus is on finishing [qmprs](https://github.com/Qualition/qmprs). Given the approximate compilation module making it to production, I will be looking forward to integrating qmprs once it reaches a minimum mature stage in UCC with the help of the supportive maintainers.
+Feel free to reach out to me via LinkedIn, or discord, or email should you have any questions or would like to collaborate.
```

Comment:
There is no email on your GitHub page, and no links to linkedin or discord handle :/